lib/SILOptimizer/Transforms/SemanticARCOpts.cpp

```diff
@@ -482,30 +483,37 @@ namespace {
 /// A simple analysis that checks if a specific def (in our case an inout
 /// argument) is ever written to. This is conservative, local and processes
 /// recursively downwards from def->use.
-struct IsAddressWrittenToDefUseAnalysis {
-  llvm::SmallDenseMap<SILValue, bool, 8> isWrittenToCache;
-
-  bool operator()(SILValue value) {
-    auto iter = isWrittenToCache.try_emplace(value, true);
-
-    // If we are already in the map, just return that.
-    if (!iter.second)
-      return iter.first->second;
-
-    // Otherwise, compute our value, cache it and return.
-    bool result = isWrittenToHelper(value);
-    iter.first->second = result;
-    return result;
-  }
-
-private:
-  bool isWrittenToHelper(SILValue value);
+struct IsAddressWrittenToDefUseAnalysis
+    : public SmallMultiMapCache<IsAddressWrittenToDefUseAnalysis, SILValue,
```

Comment:
Please explain how the Analysis is supposed to be viewed as a cache. What does the `SmallMultiMapCache` interface mean when it's inherited by this analysis. Or just don't publicly inherit a generic ADT and give the analysis its own interface.